src/I2CTransfer.cpp

```diff
@@ -66,11 +67,15 @@ uint8_t I2CTransfer::sendData(const uint16_t& messageLen, const uint8_t& packetI
 
 	numBytesIncl = packet.constructPacket(messageLen, packetID);
 
-	port->beginTransmission(targetAddress);
+	if (targetAddress < 0xFF)
+		port->beginTransmission(targetAddress);
+
 	port->write(packet.preamble, sizeof(packet.preamble));
 	port->write(packet.txBuff, numBytesIncl);
 	port->write(packet.postamble, sizeof(packet.postamble));
-	port->endTransmission();
+
+	if (targetAddress < 0xFF)
+		port->endTransmission();
 
 	return numBytesIncl;

```

Comment:
@tommag I think if the target address is invalid, the function should return 0 since nothing was transferred. Maybe we could put everything dealing with port writing, port closing, and reporting `numBytesIncl` in the first if statement and return 0 otherwise